src/mam/mod_mam_rdbms_arch_async.erl

```diff
@@ -142,6 +167,9 @@ do_flush_pm(Acc, #{host_type := HostType, batch_size := MaxSize}) ->
         {updated, _Count} -> ok;
         {error, Reason} ->
             mongoose_metrics:update(HostType, modMamDropped, MessageCount),
+            [?LOG_ERROR(#{what => archive_message_dropped,
+                          sensitive => true, params => Params})
+             || Params <- Acc],
```

Comment:
I'm afraid this will complicate GDPR, it contains private data. Don't know what is meant with the `sensitive` key, is it supposed to do some special filtering on the logger handlers or what?
Also, if anything, this is not distinguishing if the batch or the single query failed. If we did the one-by-one, it is possible that only one message failed, we don't want then to log all of them here, but only the one that failed. That is what, hopefully, `Reason` should more or less do in the log before.